### PR TITLE
Add Basic Sim Example

### PR DESCRIPTION
**Pull request by gerth2**
Building on Prateek's PR - 

Added a new example that builds upon the AimAndRange example to add sim support.

### CODE REVIEW

File: photonlib-java-examples/src/main/java/org/photonlib/examples/simaimandrange/Robot.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                // First calculate range
+                double range =
+                        PhotonUtils.calculateDistanceToTargetMeters(
+                                CAMERA_HEIGHT_METERS,
+                                TARGET_HEIGHT_METERS,
+                                CAMERA_PITCH_RADIANS,
+                                Units.degreesToRadians(result.getBestTarget().getPitch()));
+
+                // Use this range as the measurement we give to the PID controller.
+                // -1.0 required to ensure positive PID controller effort _increases_ range
```
Comment (gerth2) [inline photonlib-java-examples/src/main/java/org/photonlib/examples/simaimandrange/Robot.java:88]: Side note- this -1.0 (and a few others) probably need to be changed in the base aimandrange example - as-is, folks will have a few things reversed for the default drive base?

Comment (mcm001) [reply photonlib-java-examples/src/main/java/org/photonlib/examples/simaimandrange/Robot.java:88]: Yeah that's likely true.

**Comment by gerth2** [comment]
Will wait on #238 , put this on top, and we can review more.

**Comment by gerth2** [comment]
Docs to go along with these dudes: https://github.com/PhotonVision/photonvision-docs/pull/76

**Review by Bankst — APPROVED** [review]

**Pull request merged**